src/App.tsx

```diff
@@ -1,72 +1,101 @@
 import { SearchForm } from "./components/SearchForm/SearchForm.tsx";
 import { GenreSelect } from "./components/GenreSelect/GenreSelect.tsx";
 import { MovieTile } from "./components/MovieTile/MovieTile.tsx";
-import { useState } from "react";
+import { useEffect, useRef, useState } from "react";
+import {
+  SortControl,
+  SortOption,
+} from "./components/SortControl/SortControl.tsx";
+import { Movie } from "./models/movie.interface.ts";
+import { MovieDetails } from "./components/MovieDetails/MovieDetails.tsx";
+import { MoviesResponse } from "./models/api.ts";
 
-const movies = [
-  {
-    id: "1",
-    imageUrl: "https://example.com/pulp-fiction.jpg",
-    name: "Pulp Fiction",
-    releaseYear: 1994,
-    genres: ["Crime", "Drama"],
-  },
-  {
-    id: "2",
-    imageUrl: "https://example.com/the-shawshank-redemption.jpg",
-    name: "The Shawshank Redemption",
-    releaseYear: 1994,
-    genres: ["Drama"],
-  },
-  {
-    id: "3",
-    imageUrl: "https://example.com/the-dark-knight.jpg",
-    name: "The Dark Knight",
-    releaseYear: 2008,
-    genres: ["Action", "Crime", "Drama"],
-  },
-  {
-    id: "4",
-    imageUrl: "https://example.com/interstellar.jpg",
-    name: "Interstellar",
-    releaseYear: 2014,
-    genres: ["Adventure", "Drama", "Sci-Fi"],
-  },
-  {
-    id: "5",
-    imageUrl: "https://example.com/forrest-gump.jpg",
-    name: "Forrest Gump",
-    releaseYear: 1994,
-    genres: ["Drama", "Romance"],
-  },
-];
 const genres = ["ALL", "DOCUMENTARY", "COMEDY", "HORROR", "CRIME"];
 
 function App() {
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [selectedMovie, setSelectedMovie] = useState<Movie | null>(null);
   const [selectedGenre, setSelectedGenre] = useState(genres[0]);
+  const [sort, setSort] = useState(SortOption.ReleaseDate);
+  const [query, setQuery] = useState("Star Wars");
+
+  const abortControllerRef = useRef<AbortController | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<null | string>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      if (abortControllerRef.current) {
+        abortControllerRef.current.abort();
+      }
+      const abortController = new AbortController();
+
+      try {
+        const filter =
+          selectedGenre === "ALL" ? "" : selectedGenre.toLowerCase();
+
+        const response = await fetch(
+          `http://localhost:4000/movies?search=${query}&filter=${filter}&searchBy=title&sortBy=${sort}&sortOrder=desc`,
+          { signal: abortController.signal },
+        );
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status.toString()}`);
+        }
+        const result = (await response.json()) as MoviesResponse;
+        setMovies(result.data);
+      } catch (err) {
+        setError((err as Error).message);
+      } finally {
+        setLoading(false);
+      }
+
+      abortControllerRef.current = abortController;
```

Comment:
This won't work. You are updating your ref with new abort controller after request is done, which is too late to cancel it.